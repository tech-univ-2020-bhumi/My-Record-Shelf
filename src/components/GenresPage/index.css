.genresRows{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.genresRowOne{
    display: flex;
    flex-direction: row;
    justify-content: space-around;
}

.genresRowTwo{
    display: flex;
    flex-direction: row;
    justify-content: space-around;
}

.guitar{
    
}

.amplifier{

}

.singer{

}

.electricGuitar{

}