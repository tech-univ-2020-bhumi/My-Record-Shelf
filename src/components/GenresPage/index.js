/* eslint-disable jsx-a11y/alt-text */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import propTypes from 'prop-types';
import './index.css';
import useRecords from '../../hooks/useRecords';

const getGenresArray = (inputGenre, records) => {
  const arr = [];
  records.forEach((record) => {
    record.genres.forEach((genre) => {
      if (genre === inputGenre)arr.push(record);
    });
  });
  return arr;
};

function GenresPage(props) {
  const [records, setRecords] = useRecords();
  const { testId } = props;

  const rockArr = getGenresArray('rock', records);
  //   console.log('Rooockkk', rockArr);

  const popArr = getGenresArray('pop', records);
  //   console.log('Poppppp', popArr);

  const bollywoodArr = getGenresArray('bollywood', records);
  //   console.log('bollyyy', bollywoodArr);

  const countryArr = getGenresArray('country', records);
  //   console.log('coutrrry', countryArr);


  return (
    <div data-testid={testId}>

      <div>genres</div>

      <div className="genresRows">

        <div className="genresRowOne">

          <img src="./001-guitar.png" height="100" width="100" />
          <button type="button">rock</button>

          <img src="./049-singer.png" height="100" width="100" />
          <button type="button">pop</button>

        </div>

        <div className="genresRowTwo">

          <img src="./006-amplifier.png" height="100" width="100" />
          <button type="button">bollywood</button>

          <img src="./012-electric guitar.png" height="100" width="100" />
          <button type="button">country</button>

        </div>

      </div>
    </div>
  );
}

GenresPage.propTypes = {
  testId: propTypes.func.isRequired,
};

export default GenresPage;
