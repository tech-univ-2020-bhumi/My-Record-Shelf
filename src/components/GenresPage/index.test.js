/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import GenresPage from './index';

describe('The GenresPage component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<GenresPage testId="123" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
