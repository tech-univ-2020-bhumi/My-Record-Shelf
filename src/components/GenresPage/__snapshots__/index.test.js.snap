// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The GenresPage component should render correctly 1`] = `
<DocumentFragment>
  <div
    data-testid="123"
  />
</DocumentFragment>
`;
